os/pm/pm_initialize.c

```diff
@@ -73,6 +80,146 @@
 
 struct pm_global_s g_pmglobals;
 
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_data_handler
+ *
+ * Description:
+ *   Parse the switch, enter & exit threshold values to the global arrays.
+ *   If not provided by the SoC initilaize them with default values.
+ *
+ ****************************************************************************/
+
+static void pm_data_handler(int array_type, const char *thr_str, int threshold_array[])
+{
+	int index = 0;
+	bool assign_defaultvalue = true;
+
+	/* Initialize state threshold value to be read from thresh_str */
+	int threshold = 0;
+	/* Variable to track number of tick count threshold provided by the SOC */
+	int check_count = 0;
+
+	/* Normal state tick count threshold is NA, hence initialized to 0 */
+	threshold_array[index++] = 0;
+
+	/* Parse the sleep state tick count thresholds from the Kconfig */
+
+	while(*thr_str) {
+
+		/* Extract thresholds from string input in config as integer value */
+		threshold = strtoul(thr_str, NULL, 0);
+
+		while(*thr_str != ',' && *thr_str) {
+			thr_str++;
+		}
+
+		if(*thr_str == ',') {
+			thr_str++;
+		}
+
+		check_count++;
+
+		/* Case when more state threshold values are provided by PM_STATE_COUNT_THRESH*/
+		if(check_count > (CONFIG_PM_NSTATE - 1)) {
+			DEBUGASSERT(!(check_count > (CONFIG_PM_NSTATE - 1)));
+		}
+
+		/* Update sleep state threshold array with extracted integer value */
+		threshold_array[index++] = threshold;
+	}
+
+	/* Check if all the state entering tick count thresholds are provided by the SOC */
+
+	if(check_count == (CONFIG_PM_NSTATE - 1)) {
+	/* Case when all state threshold values are provided by PM_STATE_COUNT_THRESH*/
+
+		assign_defaultvalue = false;
+
+	} else if((check_count > 0 && check_count < (CONFIG_PM_NSTATE - 1))) {
+	/* Case when less state threshold values are provided by PM_STATE_COUNT_THRESH*/
+
+		DEBUGASSERT(!(check_count > 0 && check_count < CONFIG_PM_NSTATE));
+
+	} else {
+	/* Case when no state threshold values are provided by PM_STATE_COUNT_THRESH*/
+
+		assign_defaultvalue = true;
+
+	}
```

Comment:
I think `else` case is not needed as `assign_defaultvalue` is  initialized with `true` at the start of the function.